universal-application-tool-0.0.1/test/support/ProgramBuilder.java

```diff
@@ -26,27 +28,64 @@ private ProgramBuilder(ProgramDefinition.Builder builder) {
     this.builder = builder;
   }
 
+  public static void setInjector(Injector i) {
+    injector = i;
+  }
+
   /**
-   * Creates {@link ProgramBuilder} with a new {@link Program} with an empty name and description.
+   * Creates {@link ProgramBuilder} with a new {@link Program} with an empty name and description,
+   * in draft state.
    */
-  public static ProgramBuilder newProgram() {
-    return newProgram("", "");
+  public static ProgramBuilder newDraftProgram() {
+    return newDraftProgram("", "");
   }
 
-  /** Creates a {@link ProgramBuilder} with a new {@link Program} with an empty description. */
-  public static ProgramBuilder newProgram(String name) {
-    return newProgram(name, "");
+  /**
+   * Creates a {@link ProgramBuilder} with a new {@link Program} with an empty description, in draft
+   * state.
+   */
+  public static ProgramBuilder newDraftProgram(String name) {
+    return newDraftProgram(name, "");
   }
 
-  /** Creates a {@link ProgramBuilder} with a new {@link Program}. */
-  public static ProgramBuilder newProgram(String name, String description) {
+  /** Creates a {@link ProgramBuilder} with a new {@link Program} in draft state. */
+  public static ProgramBuilder newDraftProgram(String name, String description) {
+    VersionRepository versionRepository = injector.instanceOf(VersionRepository.class);
     Program program = new Program(name, description, name, description);
-    program.setVersion(1L);
+    program.addVersion(versionRepository.getDraftVersion());
+    maybeSave(program);
```

Comment:
Thoughts (non-blocking): this is not directly related to this change, but maybeSave is really just `save`, it wouldn't work if there is no db.